atlas-core/src/main/java/io/atlasmap/core/DefaultAtlasSession.java

```diff
@@ -94,6 +96,16 @@ public Object getInput(String docId) {
     }
 
     @Override
+    public boolean hasInput(String docId) {
+        return inputMap.containsKey(docId);
+    }
+
+    @Override
+    public Map<String, Object> getInputMap() {
```

Comment:
Do you need to expose the whole map (same for `output`) ? Actually, on a first glance I haven't found that you need it. And if you need it, maybe its better to provide a method `getInput(String docId)` in order to not expose implementation details ? (e.g. to not allow to change the map)

Reply:
Not yet right now though, IMHO very likely it'll be helpful to iterate all to support multiple input/output. Although it's just an read-only view though (Collections.unmodifiableMap), I'm fine to remove for now as it's not yet used.

Reply:
@rhuss which do you prefer? as it contains API change on AtlasSession, earlier would be better though. Yet another option is to add those only on DefaultAtlasSession for internal use only and we always cast it to DefaultAtlasSession when it's needed.

Reply:
I'm fine with exposing it in as an unmodifiable map if the chance is high that it will be needed.